test/unit/mdc-slider/foundation.test.js

```diff
@@ -0,0 +1,728 @@
+  const raf = createMockRaf();
+
+  td.when(mockAdapter.computeBoundingRect()).thenReturn({left: 0, width: 100});
+  foundation.init();
+  raf.flush();
+
+  foundation.setValue(75);
+  raf.flush();
+
+  td.verify(mockAdapter.setTrackStyleProperty(TRANSFORM_PROP, 'scaleX(0.75)'));
+
+  raf.restore();
+});
+
+test('#setValue respects the width of the slider when setting the thumb container transform', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const raf = createMockRaf();
+
+  td.when(mockAdapter.computeBoundingRect()).thenReturn({left: 0, width: 200});
+  foundation.init();
+  raf.flush();
+
+  foundation.setValue(75);
+  raf.flush();
+
+  td.verify(mockAdapter.setThumbContainerStyleProperty(TRANSFORM_PROP, 'translateX(150px) translateX(-50%)'));
+
+  raf.restore();
+});
+
+test('#setValue flips the slider thumb position across the X-axis when in an RTL context', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const raf = createMockRaf();
+
+  td.when(mockAdapter.computeBoundingRect()).thenReturn({left: 0, width: 100});
+  td.when(mockAdapter.isRTL()).thenReturn(true);
+  foundation.init();
+  raf.flush();
+
+  foundation.setValue(75);
+  raf.flush();
+
+  td.verify(mockAdapter.setThumbContainerStyleProperty(TRANSFORM_PROP, 'translateX(25px) translateX(-50%)'));
+
+  raf.restore();
+});
+
+test('#setValue adds the mdc-slider--off class when set to the minimum value', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const raf = createMockRaf();
+
+  td.when(mockAdapter.computeBoundingRect()).thenReturn({left: 0, width: 200});
+  foundation.init();
+  raf.flush();
+
+  // Change the value so that setting the value back to zero won't be short-circuited
+  foundation.setValue(10);
+  raf.flush();
+
+  foundation.setValue(0);
+  raf.flush();
+
+  // Called once on init(), once when the value is set
+  td.verify(mockAdapter.addClass(cssClasses.OFF), {times: 2});
+
+  raf.restore();
+});
+
+test('#setValue removes the mdc-slider--off class when not set to the minimum value', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const raf = createMockRaf();
+
+  td.when(mockAdapter.computeBoundingRect()).thenReturn({left: 0, width: 200});
+  foundation.init();
+  raf.flush();
+
+  // Change the value so that setting the value back to zero won't be short-circuited
+  foundation.setValue(10);
+  raf.flush();
+
+  td.verify(mockAdapter.removeClass(cssClasses.OFF));
+
+  raf.restore();
+});
+
+test('#setValue does not cause any events to be fired', () => {
+  const {foundation, mockAdapter} = setupTest();
+  const raf = createMockRaf();
+
+  td.when(mockAdapter.computeBoundingRect()).thenReturn({width: 0, left: 0});
+  foundation.init();
+  raf.flush();
+
+  foundation.setValue(20);
+
+  td.verify(mockAdapter.notifyInput(), {times: 0});
+  td.verify(mockAdapter.notifyChange(), {times: 0});
+
+  raf.restore();
+});
+
+test('#getMax/#setMax basic functionality', () => {
```

Comment:
super nit: feel the test description should be more descriptive.

How about "Slider Max getter and setter work correctly"?